lib/connection.js

```diff
@@ -548,6 +548,13 @@ Connection.prototype._ensureDB = function _ensureDB (config) {
   dbOptions.storage = config.options.storage;
   dbOptions.type = config.options.databaseType;
 
+  if ( (!config.user) && (dbOptions.username) ) {
+    // We only have database credentials; try to connect, assuming it exists.
+    return new Promise(function (resolve, reject) {
```

Comment:
I'm afraid you're wrong. You can see in line 477 (inside function _init)

    self._ensureDB(config)
      .then(...)
      ...

That is, we have another function that assumes _ensureDB is returning a promise.
Also, use() function returns a Db object, rather than a promise, as can be seen in file
    lib/server/index.js
in oriento
( e.g. https://github.com/codemix/oriento/blob/master/lib/server/index.js , line 164 )

p.s.
I will try to see why the build is breaking the tests and fix it.